### PR TITLE
Show Google Location Services dialog only once

### LINKED ISSUE #3846: Show Google Location Services dialog "For a better experience..." only once
At the moment, if user disabled location services for Google Play version, the dialog "For a better experience, turn on device location, which uses Google's location service" appears on every app activation.

It should be displayed only once per some period of time (e.g. once in 24 hours), like another "Location is disabled" dialog.

The same hide "annoying" logic should be used.

image

CC @vng

### PR DESCRIPTION
**Pull request by rtsisyk**
Closes #3846

Signed-off-by: Roman Tsisyk <roman@tsisyk.com>

### CODE REVIEW

**Review by biodranik — APPROVED** [review]

File: android/flavors/gms-enabled/app/organicmaps/location/GoogleFusedLocationProvider.java [hunk 1]
```diff
@@ -59,6 +62,7 @@ public void onLocationAvailability(@NonNull LocationAvailability availability)
     super(listener);
     mFusedLocationClient = LocationServices.getFusedLocationProviderClient(context);
     mSettingsClient = LocationServices.getSettingsClient(context);
+    mContext = context;
```
Comment (biodranik) [inline android/flavors/gms-enabled/app/organicmaps/location/GoogleFusedLocationProvider.java:65]: Can context change at some point later, so this stored one won't work properly?

**Review by vng — APPROVED** [review]

**Review by biodranik — APPROVED** [review]
It works on my devices but looks like there is another issue in cases when the location was disabled, OM started, and location was enabled again NOT via Google's dialog. The button shows "location is searched" radar icon that turns into the crossed icon immediately after a drag. While in reality, it looks like location search continues.

**Comment by rtsisyk** [comment]
> It works on my devices but looks like there is another issue in cases when the location was disabled, OM started, and location was enabled again NOT via Google's dialog. The button shows "location is searched" radar icon that turns into the crossed icon immediately after a drag. While in reality, it looks like location search continues.

Could you please provide step-by-step instructions how to reproduce this case? A lot of such issues have been fixed by #3944, this PR is unlikely to cause any new problem.

File: android/flavors/gms-enabled/app/organicmaps/location/GoogleFusedLocationProvider.java (hunk 1 above)
Comment (rtsisyk) [reply android/flavors/gms-enabled/app/organicmaps/location/GoogleFusedLocationProvider.java:65]: It is the same context which is in LocationHelper (MwmApplication).

**Pull request merged**